// @mantine/notifications/cjs/NotificationContainer/NotificationContainer.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var React = require('react');
var core = require('@mantine/core');
var getAutoClose = require('./get-auto-close/get-auto-close.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e['default'] : e; }

var React__default = /*#__PURE__*/_interopDefaultLegacy(React);

function NotificationContainer({
  className,
  sx,
  notification,
  autoClose,
  onHide,
  innerRef
}) {
  const autoCloseTimeout = getAutoClose['default'](autoClose, notification);
  const hideTimeout = React.useRef();
  const handleHide = () => {
    onHide(notification.id);
    window.clearTimeout(hideTimeout.current);
  };
  const cancelDelayedHide = () => {
    clearTimeout(hideTimeout.current);
  };
  const handleDelayedHide = () => {
    if (typeof autoCloseTimeout === "number") {
      hideTimeout.current = window.setTimeout(handleHide, autoCloseTimeout);
    }
  };
  React.useEffect(() => {
    if (typeof notification.onOpen === "function") {
      notification.onOpen(notification);
    }
  }, []);
  React.useEffect(() => {
    handleDelayedHide();
    return cancelDelayedHide;
  }, [autoClose, notification.autoClose]);
  return /* @__PURE__ */ React__default.createElement(core.Notification, {
    sx,
    className,
    title: notification.title,
    onClose: handleHide,
    color: notification.color,
    radius: notification.radius,
    icon: notification.icon,
    loading: notification.loading,
    disallowClose: notification.disallowClose,
    onMouseEnter: cancelDelayedHide,
    onMouseLeave: handleDelayedHide,
    ref: innerRef
  }, notification.message);
}
NotificationContainer.displayName = "@mantine/notifications/NotificationContainer";

exports.default = NotificationContainer;
