// @mantine/core/esm/components/AppShell/Navbar/Navbar.js
import React, { forwardRef } from 'react';
import { getDefaultZIndex, useMantineDefaultProps, Global } from '@mantine/styles';
import { useAppShellContext } from '../AppShell.context.js';
import { getSortedBreakpoints } from './get-sorted-breakpoints/get-sorted-breakpoints.js';
import { NavbarSection } from './NavbarSection/NavbarSection.js';
import useStyles from './Navbar.styles.js';
import { Box } from '../../Box/Box.js';

var __defProp = Object.defineProperty;
var __getOwnPropSymbols = Object.getOwnPropertySymbols;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __propIsEnum = Object.prototype.propertyIsEnumerable;
var __defNormalProp = (obj, key, value) => key in obj ? __defProp(obj, key, { enumerable: true, configurable: true, writable: true, value }) : obj[key] = value;
var __spreadValues = (a, b) => {
  for (var prop in b || (b = {}))
    if (__hasOwnProp.call(b, prop))
      __defNormalProp(a, prop, b[prop]);
  if (__getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(b)) {
      if (__propIsEnum.call(b, prop))
        __defNormalProp(a, prop, b[prop]);
    }
  return a;
};
var __objRest = (source, exclude) => {
  var target = {};
  for (var prop in source)
    if (__hasOwnProp.call(source, prop) && exclude.indexOf(prop) < 0)
      target[prop] = source[prop];
  if (source != null && __getOwnPropSymbols)
    for (var prop of __getOwnPropSymbols(source)) {
      if (exclude.indexOf(prop) < 0 && __propIsEnum.call(source, prop))
        target[prop] = source[prop];
    }
  return target;
};
const defaultProps = {
  fixed: false,
  position: { top: 0, left: 0 },
  zIndex: getDefaultZIndex("app"),
  hiddenBreakpoint: "md",
  hidden: false
};
const Navbar = forwardRef((props, ref) => {
  const _a = useMantineDefaultProps("Navbar", defaultProps, props), {
    width,
    height,
    fixed,
    position,
    zIndex,
    hiddenBreakpoint,
    hidden,
    className,
    classNames,
    styles,
    children
  } = _a, others = __objRest(_a, [
    "width",
    "height",
    "fixed",
    "position",
    "zIndex",
    "hiddenBreakpoint",
    "hidden",
    "className",
    "classNames",
    "styles",
    "children"
  ]);
  const ctx = useAppShellContext();
  const { classes, cx, theme } = useStyles({
    width,
    height,
    fixed: ctx.fixed || fixed,
    position,
    hiddenBreakpoint,
    zIndex: ctx.zIndex || zIndex
  }, { classNames, styles, name: "Navbar" });
  const breakpoints = getSortedBreakpoints(width, theme).reduce((acc, [breakpoint, breakpointSize]) => {
    acc[`@media (min-width: ${breakpoint + 1}px)`] = {
      "--mantine-navbar-width": `${breakpointSize}px`
    };
    return acc;
  }, {});
  return /* @__PURE__ */ React.createElement(Box, __spreadValues({
    component: "nav",
    ref,
    className: cx(classes.root, { [classes.hidden]: hidden }, className)
  }, others), children, /* @__PURE__ */ React.createElement(Global, {
    styles: () => ({
      ":root": __spreadValues({
        "--mantine-navbar-width": (width == null ? void 0 : width.base) ? `${width.base}px` : "0px"
      }, breakpoints)
    })
  }));
});
Navbar.Section = NavbarSection;
Navbar.displayName = "@mantine/core/Navbar";

export { Navbar };
